### DISCUSSION TITLE
Does LH use transfer size of resources for calculating the score under Simulated throttling mode

### DISCUSSION THREAD

**Discussion by althaf101** [GoogleChrome/lighthouse#15424] [Q&A] [answered]
I am having a react application that runs on Android Webview. I am currently intercepting all the network requests made by the webview using [ShouldInterceptRequest](https://developer.android.com/reference/android/webkit/WebViewClient#shouldInterceptRequest(android.webkit.WebView,%20android.webkit.WebResourceRequest)) method and then I depend on the native code to fetch the resource.
Now, I want to compare my react app performance with and without this interception and I am using LH (Simulated throttling) for it. 

**Observations**
1. When interception is ON, the transfer size shown in the `Network` tab in chrome DevTools is zero always. Resource size if fine.
2. Resources take longer to be fetched if interception is ON (this could be some internal perf issue in my native code). 

Still, the metrics like FCP, LCP and the total LH score is always better when Interception is ON, if not equal. So, I would like to understand whether LH is using the transfer size to calculate the LH score under simulated throttling mode, which makes the scores better for when interception is ON, even though resources load slow.

**Comment by adamraine** [comment, marked as answer]
Yes, transfer sizes are used to estimate network time under simulated throttling. If Lighthouse sees the transfer size of requests as 0 then I would expect Lighthouse to estimate better performance metrics than if the transfer size was accurate.

See also https://github.com/GoogleChrome/lighthouse/blob/main/docs/lantern.md

**Reply by althaf101** [reply]
Thanks @adamraine
